### ISSUE TITLE
Missing documentation

### ISSUE THREAD

**Issue by galinyc1** [baoguangsheng/bart-rewriters#1]
Thank you very much this repository is great! 
I was able to use your sample data but I'm not sure what should I do next.. 
Assuming I have a long text file, what are the steps to rewrite it? 
Looks like if I will just replace "test.source" content with mine, it won't be the right thing to do.

Please help me understand the steps to convert my text file.

Thanks

**Comment by baoguangsheng** [comment]
Hi, you can rewrite your own documents using our script test-rewriter.sh(https://github.com/baoguangsheng/bart-rewriters/blob/main/exp_rewriter/test-rewriter.sh). 

The only thing that you need to do is to replace the "test.source" and "test.target" files. Since you do not have a target file, you could create a dummy one with the same number of lines as the source file, but please ignore the reported ROUGE scores then. Run the script, and then you can find the logged output in folder cnndm.results.

**Comment by galinyc1** [comment]
Thank you for the quick reply, I wasn't sure since I saw files named train.source and train.target.

I did as you mentioned but it didn't work.
I placed inside exp_test\cnndm.tokenized\test.source the content I would like to rewrite
I place the same amount of (empty) lines in exp_test\cnndm.tokenized\test.target

I run the script using the same command in your README.md file:
[cmd] ➤ CUDA_VISIBLE_DEVICES=0 bash exp_rewriter/test-rewriter.sh exp_test jointsr none
Wed Oct 19 19:37:11 EDT 2022, exp_path: exp_test, data: cnndm, format: jointsr, extractor: none
Wed Oct 19 19:37:11 EDT 2022, Generate summary...
Wed Oct 19 19:37:11 EDT 2022, Calculating ROUGE...

But there is no data inside the exp_test\run-jointsr\cnndm.results output folder.
test.none.ref.tmp is 0KB
test.none.gen.tmp is 0KB
test.none.ref is 2 KB (a copy of the empty target file)

Any other suggestions? 

Thanks!

**Comment by baoguangsheng** [comment]
It works for me, so I'm wondering if there was any exception during the running. Would you please make following check? 
1) Make sure the document in test.source is tokenized and sentences are separated by " <SEP> ". (You could reference the format of our files.)
2) Check log exp_test/run-jointsr/test_none.cnndm.log to make sure there is no exception raised. (An exception could be raised if some packages in your environment were missed or uncompatible.)
3) The model output will be exp_test/run-jointsr/cnndm.results/test.none.ext and exp_test/run-jointsr/cnndm.results/test.none.gen, where .ext file is the extracted sentences and .gen file is the rewrited summary.

Hope it can solve your problem.

**Comment by galinyc1** [comment]
Sorry for the long time to response, I tried to solve it by myself, I also replaced my Windows with Linux.. 

**REGARDING #1 -** 

This is the content of the test.source:
Unlimited 2% Cash Rewards <SEP> The Wells Fargo Active Cash® Card offers unlimited 2% cash rewards on purchases. Other cards that offer higher percentages only do so on certain categories (groceries or gas, for instance) and usually limit how much you can earn in those categories. With the Wells Fargo Active Cash® Card you don’t have to worry about any caps, categories or activating your rewards since you’ll automatically earn 2% on all of your purchases. This 2% rate is tough to beat and makes this no annual fee card a stand out option for anyone with good to excellent credit that is looking for a low-maintenance cash rewards card that will allow you to accumulate cash rewards really quickly. <SEP> $200 Cash Rewards Bonus <SEP> There is also a sign-up bonus worth $200 in cash rewards that can be earned by spending at least $1,000 in the first three months after you are approved. This means you are getting another 20% back on the first $1,000 you spend. 

The content of test.target is 4 empty lines

**REGARDING #2**
The content of exp_test/run-jointsr/test_none.cnndm.log:
exp_rewriter/test-rewriter.sh: line 35: python: command not found
rouge155.sh
Mon Oct 24 10:18:23 PM EDT 2022, Calculating ROUGE...
cat: exp_test/run-jointsr/cnndm.results/test.none.gen: No such file or directory
Error: Could not find or load main class edu.stanford.nlp.process.PTBTokenizer
Caused by: java.lang.ClassNotFoundException: edu.stanford.nlp.process.PTBTokenizer
Error: Could not find or load main class edu.stanford.nlp.process.PTBTokenizer
Caused by: java.lang.ClassNotFoundException: edu.stanford.nlp.process.PTBTokenizer
exp_common/rouge155.sh: line 16: python: command not found

**REGARDING #3** 
The test.none.ext file is missing 
I have in the exp_test/run-jointsr/cnndm.results folder the following files:
test.none.gen.tmp - empty
test.none.ref - 4 empty lines
test.none.ref.tmp - empty

================ 

I ignored the errors in #2 because I think you told me to ignore

Thank you!

**Comment by baoguangsheng** [comment]
#1: Your test.source are not tokenized. You could use Standford PTBTokenizer to tokenize it first.
#2: Your log "exp_rewriter/test-rewriter.sh: line 35: python: command not found" tells that your environment is not setup properly.
#3: Since issue #2, the output files are not generated.

**Comment by galinyc1** [comment]
Regarding #2 - I thought you said to ignore ROUGE errors 

Ok, I will investigate #1 and #2 and will update here. 

Thanks

**Comment by baoguangsheng** [comment]
The error "python: command not found" has nothing to do with ROUGE. Without python, I don't know how you could even run the script.

**Comment by galinyc1** [comment]
I'm so close! 
I've missed the requirements for the model and the bert_pretrained folders
Also, I had to install python-is-python3 so the python command will be recognize. 

Now it looks like it is running but for some reason it doesn't allow the  "--batch-size 16" in the sh file (if I remove this flag it just crashes).
I'm getting the following output, any idea?

usage: rewriter.py [-h] [--no-progress-bar] [--log-interval LOG_INTERVAL]
                   [--log-format LOG_FORMAT]
                   [--tensorboard-logdir TENSORBOARD_LOGDIR] [--seed SEED]
                   [--cpu] [--tpu] [--bf16] [--memory-efficient-bf16] [--fp16]
                   [--memory-efficient-fp16] [--fp16-no-flatten-grads]
                   [--fp16-init-scale FP16_INIT_SCALE]
                   [--fp16-scale-window FP16_SCALE_WINDOW]
                   [--fp16-scale-tolerance FP16_SCALE_TOLERANCE]
                   [--min-loss-scale MIN_LOSS_SCALE]
                   [--threshold-loss-scale THRESHOLD_LOSS_SCALE]
                   [--user-dir USER_DIR] [--empty-cache-freq EMPTY_CACHE_FREQ]
                   [--all-gather-list-size ALL_GATHER_LIST_SIZE]
                   [--model-parallel-size MODEL_PARALLEL_SIZE]
                   [--checkpoint-suffix CHECKPOINT_SUFFIX]
                   [--checkpoint-shard-count CHECKPOINT_SHARD_COUNT]
                   [--quantization-config-path QUANTIZATION_CONFIG_PATH]
                   [--profile]
                   [--criterion {sentence_prediction,masked_lm,label_smoothed_cross_entropy_rewriter,label_smoothed_cross_entropy,ctc,wav2vec,composite_loss,nat_loss,sentence_ranking,legacy_masked_lm_loss,label_smoothed_cross_entropy_with_alignment,adaptive_loss,cross_entropy,vocab_parallel_cross_entropy}]
                   [--optimizer {adam,adadelta,sgd,nag,adagrad,lamb,adamax,adafactor}]
                   [--lr-scheduler {tri_stage,inverse_sqrt,reduce_lr_on_plateau,polynomial_decay,triangular,cosine,fixed}]
                   [--tokenizer {space,nltk,moses}]
                   [--bpe {byte_bpe,fastbpe,bytes,sentencepiece,subword_nmt,gpt2,bert,characters,hf_byte_bpe}]
                   [--scoring {sacrebleu,bleu,wer,chrf}] [--task TASK]
                   [--num-workers NUM_WORKERS]
                   [--skip-invalid-size-inputs-valid-test]
                   [--max-tokens MAX_TOKENS] [--batch-size BATCH_SIZE]
                   [--required-batch-size-multiple REQUIRED_BATCH_SIZE_MULTIPLE]
                   [--required-seq-len-multiple REQUIRED_SEQ_LEN_MULTIPLE]
                   [--dataset-impl DATASET_IMPL]
                   [--data-buffer-size DATA_BUFFER_SIZE]
                   [--train-subset TRAIN_SUBSET] [--valid-subset VALID_SUBSET]
                   [--validate-interval VALIDATE_INTERVAL]
                   [--validate-interval-updates VALIDATE_INTERVAL_UPDATES]
                   [--validate-after-updates VALIDATE_AFTER_UPDATES]
                   [--fixed-validation-seed FIXED_VALIDATION_SEED]
                   [--disable-validation]
                   [--max-tokens-valid MAX_TOKENS_VALID]
                   [--batch-size-valid BATCH_SIZE_VALID]
                   [--curriculum CURRICULUM] [--gen-subset GEN_SUBSET]
                   [--num-shards NUM_SHARDS] [--shard-id SHARD_ID]
                   [--distributed-world-size DISTRIBUTED_WORLD_SIZE]
                   [--distributed-rank DISTRIBUTED_RANK]
                   [--distributed-backend DISTRIBUTED_BACKEND]
                   [--distributed-init-method DISTRIBUTED_INIT_METHOD]
                   [--distributed-port DISTRIBUTED_PORT]
                   [--device-id DEVICE_ID] [--distributed-no-spawn]
                   [--ddp-backend {c10d,no_c10d}]
                   [--bucket-cap-mb BUCKET_CAP_MB] [--fix-batches-to-gpus]
                   [--find-unused-parameters] [--fast-stat-sync]
                   [--broadcast-buffers] [--distributed-wrapper {DDP,SlowMo}]
                   [--slowmo-momentum SLOWMO_MOMENTUM]
                   [--slowmo-algorithm SLOWMO_ALGORITHM]
                   [--localsgd-frequency LOCALSGD_FREQUENCY]
                   [--nprocs-per-node NPROCS_PER_NODE]
                   [--pipeline-model-parallel]
                   [--pipeline-balance PIPELINE_BALANCE]
                   [--pipeline-devices PIPELINE_DEVICES]
                   [--pipeline-chunks PIPELINE_CHUNKS]
                   [--pipeline-encoder-balance PIPELINE_ENCODER_BALANCE]
                   [--pipeline-encoder-devices PIPELINE_ENCODER_DEVICES]
                   [--pipeline-decoder-balance PIPELINE_DECODER_BALANCE]
                   [--pipeline-decoder-devices PIPELINE_DECODER_DEVICES]
                   [--pipeline-checkpoint {always,never,except_last}]
                   [--zero-sharding {none,os}] [--path PATH]
                   [--remove-bpe [REMOVE_BPE]] [--quiet]
                   [--model-overrides MODEL_OVERRIDES]
                   [--results-path RESULTS_PATH] [--beam N] [--nbest N]
                   [--max-len-a N] [--max-len-b N] [--min-len N]
                   [--match-source-len] [--no-early-stop] [--unnormalized]
                   [--no-beamable-mm] [--lenpen LENPEN] [--unkpen UNKPEN]
                   [--replace-unk [REPLACE_UNK]] [--sacrebleu]
                   [--score-reference] [--prefix-size PS]
                   [--no-repeat-ngram-size N] [--sampling]
                   [--sampling-topk PS] [--sampling-topp PS]
                   [--constraints [{ordered,unordered}]] [--temperature N]
                   [--diverse-beam-groups N] [--diverse-beam-strength N]
                   [--diversity-rate N] [--print-alignment] [--print-step]
                   [--lm-path PATH] [--lm-weight N]
                   [--iter-decode-eos-penalty N] [--iter-decode-max-iter N]
                   [--iter-decode-force-max-iter] [--iter-decode-with-beam N]
                   [--iter-decode-with-external-reranker]
                   [--retain-iter-history] [--retain-dropout]
                   [--retain-dropout-modules RETAIN_DROPOUT_MODULES [RETAIN_DROPOUT_MODULES ...]]
                   [--decoding-format {unigram,ensemble,vote,dp,bs}]
                   [--data-format {rewriter,jointsr}] [--source SOURCE]
                   [--target TARGET]
                   [--extractor {none,oracle,lead3,bertsumext,bertext}]
                   [--extractor-path EXTRACTOR_PATH]
                   [--rewriter-path REWRITER_PATH] [--outdir OUTDIR]
rewriter.py: error: argument --batch-size: invalid Optional value: '16'

**Comment by galinyc1** [comment]
I found that the issue is that I'm using Python 3.10 and not 3.7 
I won't downgrade, I will solve it myself 
Feel free to close this issue
Thanks
